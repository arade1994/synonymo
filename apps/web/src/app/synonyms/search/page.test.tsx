import { describe, expect, test, vi } from "vitest";

import mockClient, { mockGet } from "@/mocks/buildClient.mock";
import routerMock from "@/mocks/router.mock";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import SearchSynonymsPage from "./page";

vi.mock("next/navigation", () => ({
  useRouter: () => routerMock,
}));

vi.mock("@/api/buildClient", () => {
  return {
    default: () => mockClient,
  };
});

describe("SearchSynonymsPage", () => {
  test("It should render and match a snapshot by default", () => {
    const { baseElement } = render(<SearchSynonymsPage />);

    expect(baseElement).toMatchSnapshot();
  });

  test("It should render the home link", () => {
    render(<SearchSynonymsPage />);

    const homeLink = screen.getByTestId("homeLink");

    expect(homeLink).toBeDefined();
  });

  test("It should render the word input", async () => {
    render(<SearchSynonymsPage />);

    const wordInput = screen.getByTestId("wordInput");
    expect(wordInput).toBeDefined();

    await userEvent.click(wordInput);
    await userEvent.type(wordInput, "happy");
  });

  test("It should render the search button", async () => {
    render(<SearchSynonymsPage />);

    const searchButton = screen.getByTestId("searchSynonymButton");
    expect(searchButton).toBeDefined();
  });

  test("It should call the get method with the correct arguments", async () => {
    render(<SearchSynonymsPage />);

    const wordInput = screen.getByTestId("wordInput");
    const searchButton = screen.getByTestId("searchSynonymButton");

    await userEvent.click(wordInput);
    await userEvent.type(wordInput, "happy");
    await userEvent.click(searchButton);

    expect(mockGet).toHaveBeenCalledWith("/api/synonyms/happy");
  });
});
