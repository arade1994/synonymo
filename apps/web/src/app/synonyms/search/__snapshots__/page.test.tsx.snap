// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchSynonymsPage > It should render and match a snapshot by default 1`] = `
<body>
  <div>
    <main
      class="_main_8812a5"
    >
      <a
        class="_homeLink_8812a5"
        href="/"
        id="homeLink"
      >
        ‚Üê Back to Home
      </a>
      <h1
        class="_title_8812a5"
      >
        Search 
        <span
          class="_highlight_8812a5"
        >
          for
        </span>
         Synonyms
      </h1>
      <form
        class="_form_8812a5"
      >
        <div
          class="_inputGroup_8812a5"
        >
          <input
            class="_input_8812a5"
            id="wordInput"
            placeholder="Enter a word"
            type="text"
            value=""
          />
        </div>
        <button
          class="_button_8812a5"
          disabled=""
          id="searchSynonymButton"
          type="submit"
        >
          Search
        </button>
      </form>
    </main>
  </div>
</body>
`;
